### ISSUE TITLE
WebTestCase collector not special-casing return

### ISSUE THREAD

**Issue by simon-tma** [phpstan/phpstan-symfony#40]
The collector for a test client in functional tests isn't using the argument to map to the actual return type, leading to false positives.

```php
    $client = static::createClient();

    $client->enableProfiler();
    $client->request('POST', '/api/1/company);
    $this->assertSame(
      0,
      $client->getProfile()->getCollector('swiftmailer')->getMessageCount()
    );
```

This results in:
```
Call to an undefined method Symfony\Component\HttpKernel\DataCollector\DataCollectorInterface::getMessageCount()
```

**Comment by lookyman** [comment]
Thanks, I'll look into it.
